tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -30,3 +42,173 @@ class resource_type(ArmResourceManager.resource_type):
             manager_name='sqlserver',
             annotate_parent=True
         )
+
+
+class BackupRetentionPolicyFilter(Filter):
+
+    schema = type_schema(
+        'backup-retention-policy',
+        **{
+            'op': {'enum': list(scalar_ops.keys())}
+        }
+    )
+
+    @staticmethod
+    def get_sql_server_name(sql_server_id):
+        parsed = parse_resource_id(sql_server_id)
+        return parsed.get('name')
+
+    def __init__(self, operations_property, retention_limit, data, manager=None):
+        super(BackupRetentionPolicyFilter, self).__init__(data, manager)
+        self.operations_property = operations_property
+        self.retention_limit = retention_limit
+
+    def get_backup_retention_policy(self, i):
+        client = self.manager.get_client()
+        get_operation = getattr(client, self.operations_property).get
+
+        resource_group_name = i.get('resourceGroup')
+        server_name = BackupRetentionPolicyFilter.get_sql_server_name(
+            i.get(ChildResourceQuery.parent_key))
+        database_name = i.get('name')
+
+        try:
+            response = get_operation(resource_group_name, server_name, database_name)
```

Comment:
What is the difference between `.get()` and `.list_by_database()` functions? Can one db have multiple retention policies?

Reply:
I was originally using the `list_by_database` operation for both short and long term policies. For long term policies, that was returning a single instance anyways, and as far as I know, it is only possible to have 0 or 1 short term policies, so the list operation would either give a 404 (if there were none), or a list with a single element.

So for simplicity, I am just using the `get` operation in both cases now.